zeebe/logstreams/src/main/java/io/camunda/zeebe/logstreams/impl/log/SequencerMetrics.java

```diff
@@ -7,42 +7,107 @@
  */
 package io.camunda.zeebe.logstreams.impl.log;
 
-import io.prometheus.client.Histogram;
+import static io.camunda.zeebe.logstreams.impl.log.SequencerMetrics.SequencerMetricsDoc.BATCH_LENGTH_BYTES;
+import static io.camunda.zeebe.logstreams.impl.log.SequencerMetrics.SequencerMetricsDoc.BATCH_SIZE;
+
+import io.camunda.zeebe.util.micrometer.ExtendedMeterDocumentation;
+import io.camunda.zeebe.util.micrometer.MicrometerUtil.PartitionKeyNames;
+import io.micrometer.common.docs.KeyName;
+import io.micrometer.core.instrument.DistributionSummary;
+import io.micrometer.core.instrument.Meter.Type;
+import io.micrometer.core.instrument.MeterRegistry;
 
 final class SequencerMetrics {
-  private static final Histogram BATCH_SIZE =
-      Histogram.build()
-          .namespace("zeebe")
-          .name("sequencer_batch_size")
-          .help("Histogram over the number of entries in each batch that is appended")
-          .buckets(1, 2, 3, 5, 10, 25, 50, 100, 500, 1000)
-          .labelNames("partition")
-          .register();
-
-  private static final Histogram BATCH_LENGTH_BYTES =
-      Histogram.build()
-          .namespace("zeebe")
-          .name("sequencer_batch_length_bytes")
-          .help("Histogram over the size, in Kilobytes, of the sequenced batches")
-          .buckets(0.256, 0.512, 1, 4, 8, 32, 128, 512, 1024, 4096)
-          .labelNames("partition")
-          .register();
-
-  private final Histogram.Child batchSize;
-  private final Histogram.Child batchLengthBytes;
-
-  SequencerMetrics(final int partitionId) {
-    final var partitionLabel = String.valueOf(partitionId);
-    batchSize = BATCH_SIZE.labels(partitionLabel);
-    batchLengthBytes = BATCH_LENGTH_BYTES.labels(partitionLabel);
+  private final DistributionSummary batchSize;
+  private final DistributionSummary batchLengthBytes;
+
+  SequencerMetrics(final MeterRegistry meterRegistry) {
+    batchSize =
+        DistributionSummary.builder(BATCH_SIZE.getName())
+            .description(BATCH_SIZE.getDescription())
+            .serviceLevelObjectives(BATCH_SIZE.getDistributionSLOs())
+            .register(meterRegistry);
+    batchLengthBytes =
+        DistributionSummary.builder(BATCH_LENGTH_BYTES.getName())
+            .description(BATCH_LENGTH_BYTES.getDescription())
+            .serviceLevelObjectives(BATCH_LENGTH_BYTES.getDistributionSLOs())
+            .register(meterRegistry);
   }
 
   void observeBatchSize(final int size) {
-    batchSize.observe(size);
+    batchSize.record(size);
   }
 
   void observeBatchLengthBytes(final int lengthBytes) {
     final int batchLengthKiloBytes = Math.floorDiv(lengthBytes, 1024);
-    batchLengthBytes.observe(batchLengthKiloBytes);
+    batchLengthBytes.record(batchLengthKiloBytes);
+  }
+
+  @SuppressWarnings("NullableProblems")
+  public enum SequencerMetricsDoc implements ExtendedMeterDocumentation {
```

Comment:
:thought_balloon: I think the Documnetation for most of the other metrics are in a separate file, so it may be more consistent to have them in a separate file. However, in branch `stable/8.5` is already like this so it would probably create conflicts in the feature, so let's not change it

Reply:
I guess it's because it was so small :smile: I don't think it's a huge deal, imo. If it's 1 metric, it's like OK, as long as it's public.